app/models/signature.rb

```diff
@@ -177,6 +199,40 @@ def invalidate!(now = Time.current, invalidation_id = nil)
     end
   end
 
+  def anonymised?
+    anonymised_at?
+  end
+
+  def anonymise!(timestamp)
+    self.name = "Signature #{id}"
+    self.email = "signature-#{id}@example.com"
+    self.ip_address = "192.168.1.1"
+
+    if constituency_id?
+      constituency = Constituency.find_by_external_id(constituency_id)
+    else
+      constituency = nil
+    end
+
+    if constituency
+      self.postcode = constituency.example_postcode
+    else
+      self.postcode = nil
+    end
+
+    if postcode.blank? && united_kingdom?
+      # Validations require a postcode for the UK so use the NHS
+      # 'address not known' pseudo-postcode when we didn't find
+      # an example postcode for the constituency:
+      # https://en.wikipedia.org/wiki/Postcodes_in_the_United_Kingdom#ZZ99
+      self.postcode = "ZZ993WZ"
```

Comment:
@embasbm yes, because it's an attribute on the model - without the `self` it would just create a local variable called `postcode` and not be saved.